### PR TITLE
syncthing: 1.4.2 -> 1.5.0, install desktop menu link to GUI

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

Manually opening `http://127.0.0.1:8384/` in a browser is tedious, moreso if you have to talk someone else through it. Create an freedesktop application entry so that users can discover and open the Syncthing GUI page.

As far as I know we don't do this for any other services that serve a webgui, but I think its gives NixOS a more polished feel. We are able do this where other distros cannot, because we can be sure of where the GUI is bound.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/networking/syncthing/default.nix [hunk 1]
```diff
@@ -5,6 +5,7 @@ let
     buildGoModule rec {
       version = "1.4.0";
       name = "${stname}-${version}";
```
Comment (teto) [inline pkgs/applications/networking/syncthing/default.nix:7]: is name still needed ?

File: nixos/modules/services/networking/syncthing.nix [hunk 2]
```diff
@@ -429,12 +429,27 @@ in {
 
   config = mkIf cfg.enable {
 
+    environment.systemPackages = lib.optional config.xdg.menus.enable
```
Comment (teto) [inline nixos/modules/services/networking/syncthing.nix:432]: I believe makeDesktopItem runs some additionnal checks and thus should be used.

**Review by teto — CHANGES_REQUESTED** [review]

File: nixos/modules/services/networking/syncthing.nix (hunk 2 above)
Comment (ehmry) [reply nixos/modules/services/networking/syncthing.nix:432]: Right done.

File: pkgs/applications/networking/syncthing/default.nix (hunk 1 above)
Comment (ehmry) [reply pkgs/applications/networking/syncthing/default.nix:7]: Name or pname, but I don't want to modify the package any more than necesarry.

Comment (peterhoeg) [reply pkgs/applications/networking/syncthing/default.nix:7]: You might as well split it into pname and version while at it.

**Comment by peterhoeg** [comment]
1.5.0 is already in master, hence the merge conflict.

**Comment by bhipple** [comment]
`syncthing: 1.4.2 -> 1.5.0` is already done; if you'd still like to do the desktop shortcuts let's do so in another PR, as the upgrade portion here is no longer relevant.

**Pull request closed without merging**